strikt-core/src/main/kotlin/strikt/api/Assertion.kt

```diff
@@ -322,23 +361,117 @@ interface Assertion {
   }
 }
 
-private fun <Receiver, Result> (Receiver.() -> Result).describe(): String =
+private val DESCRIBE_CACHE = ConcurrentHashMap<Class<*>, () -> String>()
+
+private fun <Receiver, Result> (Receiver.() -> Result).describe(): () -> String =
   when (this) {
-    is KProperty<*> ->
-      "value of property $name"
-    is KFunction<*> ->
-      "return value of $name"
-    is CallableReference -> "value of $propertyName"
+    is KProperty<*> -> {
+      { "value of property $name" }
+    }
+
+    is KFunction<*> -> {
+      { "return value of $name" }
+    }
+
+    is CallableReference -> {
+      { "value of $propertyName" }
+    }
+
     else -> {
-      try {
-        val line = FilePeek.filePeek.getCallerFileInfo().line
-        LambdaBody("get", line).body.trim()
-      } catch (e: Exception) {
-        "%s"
+      var lambda = DESCRIBE_CACHE[javaClass]
```

Comment:
The second commit is some further experimentation in caching the describe string calculation. I don't know if it is guaranteed behavior (I can't find any documentation), but the class created for each lambdas is unique for the location of the lambda. In my `get perf` test, the class names are something like `"get perf"$1$1`, `"get perf"$1$2`, and so on.

Adding a cache moves the example test from 1.1s to 200ms.